components/brave_rewards/browser/rewards_notification_service_impl.cc

```diff
@@ -74,10 +74,19 @@ void RewardsNotificationServiceImpl::AddNotification(
 
 void RewardsNotificationServiceImpl::DeleteNotification(RewardsNotificationID id) {
   DCHECK(!id.empty());
-  if (rewards_notifications_.find(id) == rewards_notifications_.end())
-    return;
-  RewardsNotification rewards_notification = rewards_notifications_[id];
-  rewards_notifications_.erase(id);
+  RewardsNotification rewards_notification;
+  if (rewards_notifications_.find(id) == rewards_notifications_.end()) {
+    rewards_notification.id_ = id;
+    rewards_notification.type_ = RewardsNotificationType::REWARDS_NOTIFICATION_INVALID;
+
+    // clean up, so that we don't have long standing notifications
+    if (rewards_notifications_.size() == 1) {
```

Comment:
What if we have more than one? Should we clear out the invalid?

Reply:
problem is that we only know, for sure, just for one. So we can only clear one, if there are more then one we don't know which on is invalid and which one is valid

Reply:
This seems problematic to me because it could delete the wrong notification. It feels like we're putting a bandaid over the real problem. Why is the id not correct in the first place? Where does it get lost?

Reply:
or how do we end up with an invalid id in the first place?

Reply:
we had multiple changes to the notification ID and not all migration paths were covered. We also had this problem https://github.com/brave/brave-browser/issues/2929